constructor/conda_interface.py

```diff
@@ -90,17 +90,28 @@ def get_repodata(url):
     def write_repodata(cache_dir, url, full_repodata, used_packages):
         used_repodata = {k: full_repodata[k] for k in
                          set(full_repodata.keys()) - {'packages', 'packages.conda', 'removed'}}
-        repodata_filename = _cache_fn_url(used_repodata['_url'].rstrip("/"))
         used_repodata['packages.conda'] = {}
         used_repodata['removed'] = []
-        # arbitrary old, expired date, so that conda will want to immediately update it
-        # when not being run in offline mode
-        used_repodata['_mod'] = "Mon, 07 Jan 2019 15:22:15 GMT"
         used_repodata['packages'] = {
             k: v for k, v in full_repodata['packages'].items() if v['name'] in NAV_APPS}
         for package in used_packages:
             for key in ('packages', 'packages.conda'):
                 if package in full_repodata.get(key, {}):
                     used_repodata[key][package] = full_repodata[key][package]
-        with open(join(cache_dir, repodata_filename), 'w') as fh:
-            json.dump(used_repodata, fh, indent=2)
+
+        # The first line of the JSON should contain cache metadata
+        # Choose an arbitrary old, expired date, so that conda will want to
+        # immediately update it when not being run in offline mode
+        url = used_repodata.pop('_url').rstrip("/")
+        repodata = json.dumps(used_repodata, indent=2)
+        repodata_header = json.dumps(
+            {
+                "_mod": "Mon, 07 Jan 2019 15:22:15 GMT",
+                "_url": url,
+                "_cache_control": "public, max-age=1200",
```

Comment:
it turns out that conda also only writes two of these keys (_mod and _url) when fetching any file from a local channel.

I am adding some handling in mamba to make sure that we can also parse that correctly.

Reply:
Ah great, I guess that makes sense for local channels. Thanks for dealing with this!

In that case the only thing left for this PR would be to make sure that these keys are on the top line of the file? i.e.

```python
{"_mod": "Mon, 07 Jan 2019 15:22:15 GMT", "_url": url,
  "packaes": {
       ...
  },
  "packaes.conda": {
       ...
  }
}
```

Reply:
yes, mamba currently expect them at the beginning of the file. Is that not the case here?

Reply:
Prior to this PR, constructor just writes `json.dumps(repodata, indent=2)`.

Conda does it in a way that's consistent with `mamba`.

Reply:
yeah i don't like that ... in that case we'd need to parse the entire file first. Would be nice to correct this in constructor :)

Reply:
I've amended this so that the `_cache_control` and `_etag` fields aren't written.